packages/subiquity_client/pubspec.yaml

```diff
@@ -4,19 +4,17 @@ version: 0.0.1
 homepage: https://github.com/canonical/ubuntu-desktop-installer
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.12.0 <3.0.0'
 
 dependencies:
-  diacritic:
-  tuple:
-  http: ^0.12.2
-  freezed_annotation: ^0.12.0
-  json_annotation: ^3.1.1
+  freezed_annotation: ^0.14.1
+  http: ^0.13.1
+  json_annotation: ^4.0.1
   path: ^1.8.0
 
 dev_dependencies:
-  effective_dart: ^1.3.0
-  build_runner: ^1.11.5
-  freezed: ^0.12.7
-  json_serializable: ^3.5.1
-  test: ^1.16.0
+  build_runner: ^2.0.1
+  effective_dart: ^1.3.1
+  freezed: ^0.14.1
+  json_serializable: ^4.1.1
+  test: ^1.17.0
```

Comment:
Sorry for the re-ordering here. I'm using the Pubspec Assistant VS Code extension for managing dependencies. It likes to keep these in alphabetical order - I hope that's ok...